src/Model/Table/SentencesListsTable.php

```diff
@@ -155,33 +155,44 @@ public function getUserChoices($userId, $sentenceId)
             ->notMatching('SentencesSentencesLists', function ($q) use ($sentenceId) {
                 return $q->where(['SentencesSentencesLists.sentence_id' => $sentenceId]);
             })
-            ->select(['id', 'name', 'user_id'])
-            ->order(['name']);
-
-        $listsOfUser = array();
-        $collaborativeLists = array();
-
-        $currentUserId = CurrentUser::get('id');
-        foreach ($results as $result) {
-            $listId = $result['id'];
-            $listName = $result['name'];
-            $userId = $result['user_id'];
+            ->select([
+                'id',
+                'name',
+                'user_id',
+                'is_mine' => 'IF(SentencesLists.user_id = '.$userId.', TRUE, FALSE)',
+                'is_collaborative' => 'IF(SentencesLists.editable_by = "anyone", TRUE, FALSE)',
+            ]);
 
-            if (empty($listName)) {
-                $listName = __('unnamed list');
+        if ($forNewDesign) {
+            $results->order(['is_mine DESC', 'modified DESC']);
+            return $results->toList();
+        } else {
+            $results->order(['name']);
+            $listsOfUser = array();
+            $collaborativeLists = array();
+    
+            $currentUserId = CurrentUser::get('id');
```

Comment:
I know that this is not an addition from you (existing code just got indented), but I think we should use `$userId` provided as parameter instead of calling CurrentUser here.